### PR TITLE
VZ-6467 Clean up managed cluster resources on VMC deletion

### PR DESCRIPTION
**Pull request by davisgomes**

### CODE REVIEW

File: application-operator/mcagent/mcagent_delete.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+func (s *Syncer) syncMCAgentDeleteResources() error {
+	vmcName := client.ObjectKey{Name: s.ManagedClusterName, Namespace: constants.VerrazzanoMultiClusterNamespace}
+	vmc := platformopclusters.VerrazzanoManagedCluster{}
+	err := s.AdminClient.Get(s.Context, vmcName, &vmc)
+	if client.IgnoreNotFound(err) != nil && !apierrors.IsUnauthorized(err) {
+		s.Log.Errorf("Failed to get the VMC resources %s/%s from the admin cluster: %v", constants.VerrazzanoMultiClusterNamespace, s.ManagedClusterName, err)
+		return err
+	}
+	if err == nil && vmc.DeletionTimestamp.IsZero() {
+		s.Log.Debugf("VMC resource %s/%s has been found and is not being deleted, skipping the MC Agent deletion process")
```
Comment (desagar) [inline application-operator/mcagent/mcagent_delete.go:26]: it's not really "MC Agent deletion" - same comment as above - we should call it deregistration

File: application-operator/mcagent/mcagent.go [hunk 2]
```diff
@@ -124,6 +124,13 @@ func (s *Syncer) ProcessAgentThread() error {
 	// Sync multi-cluster objects
 	s.SyncMultiClusterResources()
 
+	// Delete the managed cluster resources if de-registration occurs
+	err = s.syncMCAgentDeleteResources()
```
Comment (desagar) [inline application-operator/mcagent/mcagent.go:128]: I'd prefer it was called syncDeregister() - "Resources" is overused and not clear that it means registration-specific resources.
Same with the log message below - failed to sync deregistration is a better msg. Likewise the comment as well.

File: application-operator/mcagent/mcagent_delete.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+		return nil
+	}
+
+	s.Log.Infof("Verrazzano Managed Cluster %s/%s has been deleted, cleaning up managed cluster resources", constants.VerrazzanoMultiClusterNamespace, s.ManagedClusterName)
```
Comment (fredtibbitts) [inline application-operator/mcagent/mcagent_delete.go:30]: Will the mcagent loop keep running long after the VMC is deleted? If so, you might want to make this a `Oncef` to keep the log sparse.

File: application-operator/mcagent/mcagent_delete.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022, Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package mcagent
+
+import (
+	ctx "context"
```
Comment (fredtibbitts) [inline application-operator/mcagent/mcagent_delete.go:7]: Is it necessary to alias this package? Not aliasing is preferred unless it's aliased due to a conflict. Also missing a blank link after standard lib imports.

Comment (davisgomes) [reply application-operator/mcagent/mcagent_delete.go]: I will update this.

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**